### PR TITLE
dev_env: binaries auto download, custom configs for services

### PR DESCRIPTION
**Pull request by evgeniiz321**

### CODE REVIEW

File: src/neofs_testlib/env/env.py [hunk 1]
```diff
@@ -199,23 +202,69 @@ def persist(self) -> str:
...
-        neofs_env = NeoFSEnv()
+    def simple(cls, neofs_env_config: dict = None) -> "NeoFSEnv":
+        if not neofs_env_config:
+            neofs_env_config = yaml.safe_load(files("neofs_testlib.env.templates").joinpath("neofs_env_config.yaml").read_text())
+        neofs_env = NeoFSEnv(neofs_env_config=neofs_env_config)
+        neofs_env.download_binaries()
         neofs_env.deploy_inner_ring_node()
         neofs_env.deploy_storage_nodes(
-            count=4, 
+            count=4,
```
Comment (roman-khimov) [inline src/neofs_testlib/env/env.py:264]: Mixing whitespace changes with real ones.

File: src/neofs_testlib/env/env.py [hunk 2]
```diff
@@ -199,23 +202,69 @@ def persist(self) -> str:
...
             node_attrs={
                 0: ["UN-LOCODE:RU MOW", "Price:22"],
                 1: ["UN-LOCODE:RU LED", "Price:33"],
                 2: ["UN-LOCODE:SE STO", "Price:11"],
-                3: ["UN-LOCODE:FI HEL", "Price:44"]
-            }
+                3: ["UN-LOCODE:FI HEL", "Price:44"],
+            },
```
Comment (roman-khimov) [inline src/neofs_testlib/env/env.py:270]: Same here.

File: src/neofs_testlib/env/env.py [hunk 3]
```diff
@@ -199,23 +202,69 @@ def persist(self) -> str:
...
+        binaries = [
+            (self.neofs_adm_path, "neofs_adm"),
+            (self.neofs_cli_path, "neofs_cli"),
+            (self.neo_go_path, "neo_go"),
+            (self.neofs_ir_path, "neofs_ir"),
+            (self.neofs_node_path, "neofs_node"),
+            (self.neofs_s3_authmate_path, "neofs_s3_authmate"),
+            (self.neofs_s3_gw_path, "neofs_s3_gw"),
+            (self.neofs_rest_gw_path, "neofs_rest_gw"),
+            (self.neofs_http_gw_path, "neofs_http_gw"),
```
Comment (roman-khimov) [inline src/neofs_testlib/env/env.py:220]: Why underscores in binary names? They're using dashes in fact.

File: src/neofs_testlib/env/templates/neofs_env_config.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+binaries:
+    neofs_cli:
+        repo: 'nspcc-dev/neofs-node'
+        version: 'v0.39.2'
```
Comment (roman-khimov) [inline src/neofs_testlib/env/templates/neofs_env_config.yaml:4]: Can be 0.40.1 now.

File: src/neofs_testlib/env/env.py [hunk 5]
```diff
@@ -427,11 +492,11 @@ def start(self, fresh=True):
         self._launch_process()
         logger.info(f"Wait until storage node is READY")
         self._wait_until_ready()
-        
+
```
Comment (roman-khimov) [inline src/neofs_testlib/env/env.py:522]: Whitespace.

File: src/neofs_testlib/env/env.py [hunk 6]
```diff
@@ -427,11 +492,11 @@ def start(self, fresh=True):
...
     @allure.step("Stop storage node")
     def stop(self):
         self.process.terminate()
-        
+
```
Comment (roman-khimov) [inline src/neofs_testlib/env/env.py:526]: Should not.

File: src/neofs_testlib/env/env.py [hunk 7]
```diff
@@ -452,7 +517,7 @@ def delete_data(self):
             state_file=self.state_file,
         )
         time.sleep(1)
-        
+
```
Comment (roman-khimov) [inline src/neofs_testlib/env/env.py:552]: Be mixed.

File: src/neofs_testlib/env/env.py [hunk 8]
```diff
@@ -468,12 +533,14 @@ def delete_metadata(self):
...
```
Comment (roman-khimov) [inline src/neofs_testlib/env/env.py:573]: With real changes.

File: src/neofs_testlib/env/env.py [hunk 9]
```diff
@@ -206,7 +207,7 @@ def persist(self) -> str:
     def download_binaries(self):
         logger.info(f"Going to download missing binaries, if any")
         deploy_threads = []
-        
+
```
Comment (roman-khimov) [inline src/neofs_testlib/env/env.py:210]: :eyes:

File: src/neofs_testlib/env/templates/neofs_env_config.yaml [hunk 10]
```diff
@@ -29,7 +29,7 @@ binaries:
         version: 'v0.29.0'
         file: 'neofs-s3-gw-linux-amd64'
 
-    neofs_s3_gw_authmate:
+    neofs_s3_authmate:
```
Comment (roman-khimov) [inline src/neofs_testlib/env/templates/neofs_env_config.yaml:32]: It's a different fix.

**Review by roman-khimov — COMMENTED** [review]
Please pay attention to commit structure.

Will I be able to provide some local binary?

File: src/neofs_testlib/env/env.py (hunk 3 above)
Comment (evgeniiz321) [reply src/neofs_testlib/env/env.py:220]: these are not binary names, these are keys from config file that leads to actual data

File: src/neofs_testlib/env/templates/neofs_env_config.yaml (hunk 4 above)
Comment (evgeniiz321) [reply src/neofs_testlib/env/templates/neofs_env_config.yaml:4]: It was all tested with 0.39.2. I will update versions to current ones in a separate change during tests sync with actual node version.

File: src/neofs_testlib/env/templates/neofs_env_config.yaml [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+    neofs_node:
+        repo: 'nspcc-dev/neofs-node'
+        version: 'v0.39.2'
+        file: 'neofs-node-amd64'
+
+    neofs_s3_gw:
+        repo: 'nspcc-dev/neofs-s3-gw'
+        version: 'v0.29.0'
+        file: 'neofs-s3-gw-linux-amd64'
```
[the remark this answers is not in the document]
Comment (evgeniiz321) [reply src/neofs_testlib/env/templates/neofs_env_config.yaml:30]: This is a file name from Assets dir - https://github.com/nspcc-dev/neofs-s3-gw/releases/tag/v0.29.0 - https://github.com/nspcc-dev/neofs-s3-gw/releases/download/v0.29.0/neofs-s3-gw-linux-amd64

File: src/neofs_testlib/env/templates/neofs_env_config.yaml (hunk 10 above)
Comment (evgeniiz321) [reply src/neofs_testlib/env/templates/neofs_env_config.yaml:32]: Updated commit structure, now is better

**Comment by evgeniiz321** [comment]
> Please pay attention to commit structure.
> 
> Will I be able to provide some local binary?

Updated commit structure, now should be better. 
For local binaries: if there is already a binary in the local dir, new one will not be downloaded, so we can easily replace it with what we wanted.

File: src/neofs_testlib/env/templates/neofs_env_config.yaml (hunk 11 above)
[the remark this answers is not in the document]
Comment (roman-khimov) [reply src/neofs_testlib/env/templates/neofs_env_config.yaml:30]: I mean you can have $OS and $ARCH and neofs-s3-gw-$OS-$ARCH of some kind.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**